src/test_payload_dataset_manager.cc

```diff
@@ -137,4 +134,88 @@ TEST_CASE("PayloadDatasetManager::GroupPayloadsBySession")
   }
 }
 
+TEST_CASE("PayloadDatasetManager::GetDelayForPayload")
+{
+  const std::string name{"delay"};
+  const std::string datatype{"UINT64"};
+  const std::vector<int64_t> shape{{1}};
+  const bool is_shape_tensor{false};
+  const bool is_optional{true};
+
+  const ModelTensor delay_model_tensor{
+      name, datatype, shape, is_shape_tensor, is_optional};
+
+  auto model_parser{std::make_shared<ModelParser>()};
+
+  (*model_parser->Inputs())[name] = delay_model_tensor;
+
+  const std::string input_data_json{R"(
+      {
+        "data": [
+          {
+            "delay": [1000]
+          }
+        ]
+      }
+      )"};
+
+  auto data_loader{std::make_shared<MockDataLoader>()};
+
+  const auto error{data_loader->ReadDataFromJSON(
+      model_parser->Inputs(), model_parser->Outputs(), input_data_json)};
+
+  REQUIRE(error.IsOk());
+
+  const PayloadDatasetManager payload_dataset_manager(
+      data_loader, model_parser);
+
+  const size_t dataset_index{0};
+
+  const auto delay{payload_dataset_manager.GetDelayForPayload(dataset_index)};
+
+  CHECK(delay == std::chrono::milliseconds(1000));
+}
+
+TEST_CASE("PayloadDatasetManager::GetPayload")
```

Comment:
Similarly there was no test for this public method, so I added one.

It fetches a payload out of the DataLoader via the PayloadDatasetManager class.